mindmeld/components/entity_resolver.py

```diff
@@ -466,8 +558,78 @@ def _predict(self, nbest_entities, allowed_cnames=None):
         """
         raise NotImplementedError
 
+    def dump(self, path, cache_kb=True):
+        """
+        Dumps the state of the entity resolver. The state for embedder model is the cached
+        embeddings whereas for text features based resolvers, (if required,) it will generally be a
+        serialized pickle of the underlying model/algorithm and the data associated.
+
+        In general, this method leads to creation of three files:
+            - .pkl: pickle of the underlying model/algo state (not required for Elasticsearch)
+            - .configs.pkl: pickle of the resolver's configuarble parameters
+            - .pkl.hash: a hash string obtained from a combination of KB data and the config params
+
+        Args:
+            path (str): A .pkl file path where the resolver will be dumped. The model hash will be
+                dumped at {path}.hash file path.
+            cache_kb (bool, default: True): If True, saves the processed entity map in disk and
+                loads it back when the load method is called. If False, the resolver model/algorithm
+                is dumped but not the processed entity map necessitating the need to manualy set it
+                upon loading (useful in cases like NativeQuestionAnswerer!)
+        """
+
+        # dump underlying resolver model/algorithm/embeddings
+        self._dump(path)
```

Comment:
Optional: Maybe we should add a message to the user consistent with other models - "Saving entity resolver: domain=<>, intent=<>, entity=<>"